letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+
+:tada: Congratulations, that's step 2 of 3 completed for our client! Now switch back to the terminal where you're running the shop and let's see if the payment has arrived...
+
+It hasn't, has it? So let's figure out why.
+
+### Finding your transfer on the ledger
+
+> The following short section is very specific to XRP Ledger but gives you an idea of what to do if you are debugging a failed payment. If you would rather skip ahead to the next section you can.
+
+Let's dissect what should have happened under the hood. You used the `pay.js` script to load the XRP Escrow plugin and prepare an escrow payment for 10 Drops (0.00001 XRP) to the address provided by the shop. 
+
+Our payment hasn't arrived at the shop so let's start by inspecting the XRP Testnet Ledger and see if that payment was prepared. We're going to use the REST API exposed by the same testnet server we connect our plugin to.
+
+First you need to find the sending account address you used. Look in `plugins.js` and find the account you configured your customer plugin with, or look at the output from running `pay.js` (you only need the Ripple address so trim off the ILP ledger prefix).
+
+The following will install a small tool that formats the JSON output to make it easier to read:
+
+```shell
+npm install -g jslint
+```
+
+Next use `curl` to POST an API call to the XRP Testnet server (replace the address with your sending address):
+
+```shell
+curl -X POST -d '{ "method": "account_objects", "params": [{"ledger_index":"validated", "account": "YOUR-SENDING-ADDRESS", "type": "escrow"}]}' https://client.altnet.rippletest.net:51234
+```
+
+*Installing `curl` is out of scope for this tutorial so if you don't already have it you can skip ahead or figure that bit out yourself.*
+
+You should get back a JSON object that shows all the transactions sent from your account. In among them should be the transaction you just created. 
+
+So our investigation has revealed that the transfer has been created on the ledger. We need to go back to our shop service and figure out why it wasn't received.
+
+## Step 4: Accepting the payment
+
+If you look through what we've got in `shop.js` so far you'll notice a comment `//Handle incoming transfers...`. That's our first hint that we're missing some code.
+
+This is our first introduction to some of the events that can be raised by ledger plugins. There are a number of these and the one we are interested in is called `incoming_prepare`. This event is raised whenever a transfer is prepared on the ledger and your plugin's account is the intended recipient.
+
+As soon as yur plugin connects to the ledger it will start listening for incoming transfers and should raise this event every time one occurs.
+
+Replace `//Handle incoming transfers...` with the following code:
+
+```js
+  plugin.on('incoming_prepare', function (transfer) {
+
+    if (transfer.amount !== '10') {
```

Comment:
actually, it should only through InsufficientSourceAmount if `parseInt(transfer.amount) < 10`